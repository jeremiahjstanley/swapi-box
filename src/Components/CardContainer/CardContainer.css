.card-container {
  display: flex;
  justify-content: space-evenly;
  flex-direction: row;
  flex-wrap: wrap
}

.card-container p {
  color: #fff;
  font-family: starjedi;
  font-size: 3rem;
}