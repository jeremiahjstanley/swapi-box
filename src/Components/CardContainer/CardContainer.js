import React from 'react';
import './CardContainer.css';
import PropTypes from 'prop-types';
import Card from '../Card/Card';

const CardContainer = ({state, cardType, addToFavorites, removeFromFavorites}) => {

  const containerCards = state[cardType].map((card, index) => {
    return (
      <Card
        card={card}
        key={index}
        addToFavorites={addToFavorites}
        removeFromFavorites={removeFromFavorites}
      />
    );
  });

  if (state.error.name) {
    return (
      <div className='card-container'>
        <h2 className='error-header'>{state.error.name}</h2>
        <p className='error-message'>{state.error.message}</p>
      </div>
    );
  }

  if (state[cardType].length) {
    return (
      <div className='card-container full'>
        {containerCards}
      </div>
    );
  }

  if (cardType === 'favorites') {
    return (
      <div className='card-container'>
        <p className='favorites-error'>
        You currently have no favorites
        </p>
      </div>
    );
  }

  else 
    return (
      <div className='card-container'>
        <p className='instructions'>Click a button</p>
      </div>
    );
};

CardContainer.propTypes = {
  state: PropTypes.object.isRequired,
  cardType: PropTypes.string.isRequired,
  error: PropTypes.object,
  addToFavorites: PropTypes.func.isRequired,
  removeFromFavorites: PropTypes.func.isRequired
};

export default CardContainer;