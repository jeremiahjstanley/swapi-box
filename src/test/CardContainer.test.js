import React from 'react';
import { shallow } from 'enzyme';
import CardContainer from '../Components/CardContainer/CardContainer';

describe('CardContainer test', () => {
  let mockCards;
  let mockDisplayFavorites;
  let mockError;
  let wrapper;
  let mockAddToFavorites;
  let mockRemoveFromFavorites;


  beforeEach(() => {
    mockCards = [{}, {}, {}];
    mockError = {};
    mockDisplayFavorites = false;
    mockAddToFavorites = jest.fn()
    mockRemoveFromFavorites = jest.fn()
    wrapper = shallow(<CardContainer 
                        cards={mockCards}
                        error={mockError}
                        displayFavorites={mockDisplayFavorites}
                        addToFavorites={mockAddToFavorites}
                        removeFromFavorites={mockRemoveFromFavorites}
                      />);
  })

  it('should match the snapshot',() => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should render a message if the user has no favorites',() => {
    mockCards = [];
    mockDisplayFavorites = true;
    wrapper = shallow(<CardContainer 
                        cards={mockCards}
                        error={mockError}
                        displayFavorites={mockDisplayFavorites}
                        addToFavorites={mockAddToFavorites}
                        removeFromFavorites={mockRemoveFromFavorites}
                      />);

    const expected = 1;
    const actual = wrapper.find('.favorites-error').length;

    expect(expected).toEqual(actual);
  })

  it('should render a error message if there is an error',() => {
    mockError = {name: 'error name', message: 'error message'};
    wrapper = shallow(<CardContainer 
                        cards={mockCards}
                        error={mockError}
                        displayFavorites={mockDisplayFavorites}
                        addToFavorites={mockAddToFavorites}
                        removeFromFavorites={mockRemoveFromFavorites}
                      />);

    const expected = 1;
    const actual = wrapper.find('.error-header').length;

    expect(expected).toEqual(actual);
  })

  it('should render instructions on page load',() => {
    mockCards = [];
    wrapper = shallow(<CardContainer 
                        cards={mockCards}
                        error={mockError}
                        displayFavorites={mockDisplayFavorites}
                        addToFavorites={mockAddToFavorites}
                        removeFromFavorites={mockRemoveFromFavorites}
                      />);

    const expected = 1;
    const actual = wrapper.find('.instructions').length;

    expect(expected).toEqual(actual);
  })

})