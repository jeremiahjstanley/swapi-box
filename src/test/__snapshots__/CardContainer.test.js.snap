// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer test should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    addToFavorites={[Function]}
    cards={
        Array [
            Object {},
            Object {},
            Object {},
          ]
    }
    displayFavorites={false}
    error={Object {}}
    removeFromFavorites={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          addToFavorites={[Function]}
          card={Object {}}
          removeFromFavorites={[Function]}
/>,
        <Card
          addToFavorites={[Function]}
          card={Object {}}
          removeFromFavorites={[Function]}
/>,
        <Card
          addToFavorites={[Function]}
          card={Object {}}
          removeFromFavorites={[Function]}
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "addToFavorites": [Function],
          "card": Object {},
          "removeFromFavorites": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "addToFavorites": [Function],
          "card": Object {},
          "removeFromFavorites": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "addToFavorites": [Function],
          "card": Object {},
          "removeFromFavorites": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            addToFavorites={[Function]}
            card={Object {}}
            removeFromFavorites={[Function]}
/>,
          <Card
            addToFavorites={[Function]}
            card={Object {}}
            removeFromFavorites={[Function]}
/>,
          <Card
            addToFavorites={[Function]}
            card={Object {}}
            removeFromFavorites={[Function]}
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "addToFavorites": [Function],
            "card": Object {},
            "removeFromFavorites": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "addToFavorites": [Function],
            "card": Object {},
            "removeFromFavorites": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "addToFavorites": [Function],
            "card": Object {},
            "removeFromFavorites": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
